sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/text/TextOutputWriter.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.execution.datasources.text
+
+import java.io.OutputStream
+
+import org.apache.hadoop.fs.Path
+import org.apache.hadoop.mapreduce.TaskAttemptContext
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.datasources.{CodecStreams, OutputWriter}
+import org.apache.spark.sql.types.StructType
+
+class TextOutputWriter(
+    path: String,
+    dataSchema: StructType,
+    lineSeparator: Array[Byte],
+    context: TaskAttemptContext)
```

Comment:
Shall we switch the parameter order?
```scala
-    lineSeparator: Array[Byte],
-    context: TaskAttemptContext)
+    context: TaskAttemptContext,
+    lineSeparator: Array[Byte])
```

Reply:
BTW, +1 for the decision to pass `lineSeparator` instead of full `TextOptions`.

Reply:
@dongjoon-hyun Thanks for the review. I have addressed other comments. This one is kind of trivial. I prefer to keep the original parameter order since it is not relevant to this PR.

Reply:
Got it.